packages/hash/api/src/model/index.ts

```diff
@@ -13,6 +13,18 @@ import Page from "./page.model";
 import Block from "./block.model";
 import Link from "./link.model";
 import Aggregation from "./aggregation.model";
+import DataType from "./ontology/dataType.model";
+import PropertyType from "./ontology/propertyType.model";
+import LinkType from "./ontology/linkType.model";
+
+export * from "./ontology/dataType.model";
+export { DataType };
```

Comment:
Why do we need to export DataType if we're already doing `export * from` the file?

Reply:
(and the same for the ones below)

Reply:
Is this just because of a default export and weird shadowing? If so this makes me even more confident that all this aliasing is not ideal

Reply:
I'm not sure about the specific reason for why this has to be. @benwerner01 would be the right person to answer. But it had something to do with ensuring import order is correct when using Model classes across other model classes. I'm sure he can give a better explanation about it :)

Reply:
(Had to remind myself by finding [the commit](https://github.com/hashintel/dev/pull/193/commits/e59a9345db1603409c9d8fb1496660b374afd4c4)), but the reason for this was to trick code editors into importing `ClassName` from the `model/index.ts` file, rather than the `model/className.model.ts` when auto-completing import statements.

There might be a better way to accomplish this, but at the time I was not able to come up with one.

Reply:
At the very least can we please add a comment saying that? I don't think we should be dropping in magic weird unintuitive import and export orderings to accomplish some IDE behaviour, and then just leaving it there to be broken later on.

Also I don't quite understand how this will happen if you have `__ClassName` inside the model file, because if I try and `import ClassName` surely it will only find the one in `index.ts`? Unless it's because it's then reimported in the child from the parent in some weird circular import.

This feels like it should be solved through visibility though, where only the one in index.ts is public, but that's mostly my rust-side thinking

Reply:
> weird unintuitive import and export orderings to accomplish some IDE behaviour

Just to be clear the enforced import/export orderings prevent run-time cyclic dependency errors, and don't have anything to with improving IDE behaviours.

Reply:
We might have found a way to get around this with `@internal` jsdoc markers. We'll see

Reply:
> prevent run-time cyclic dependency errors

Is this because we have literal cyclic imports though? Like the child is reimporting one of its own members from the parent

Reply:
The dependency cycle can occur if the child extends its parent and the parent imports the child. For example the `Org` model class extends the `Account` model class, therefore imports it, and the `Account` model class imports the `Org` model class, because it uses it in one of  its methods. You can find the more detailed explanation of this issue in [this internal thread](https://hashintel.slack.com/archives/C022217GAHF/p1631000961128300?thread_ts=1630928066.121100&cid=C022217GAHF).

Reply:
A comment was added as part of https://github.com/hashintel/hash/pull/894/commits/bebeae6bc0e2050b6251db7dce51b1b91875ad3a to make the purpose of the `model/index.ts` file more explicit.